### PR TITLE
Bump system io file system

### PR DESCRIPTION
**Pull request by JeremyKuhne**
@weshaggard, @ericstj, @danmosemsft 

Goes with #9736

### CODE REVIEW

File: src/System.IO.FileSystem/pkg/System.IO.FileSystem.pkgproj [hunk 1]
```diff
@@ -2,12 +2,18 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
+    <ProjectReference Include="..\ref\4.0.2\System.IO.FileSystem.depproj">
```
Comment (weshaggard) [inline src/System.IO.FileSystem/pkg/System.IO.FileSystem.pkgproj]: @ericstj can confirm but I think we reversion this to 4.0.1 as we haven't shipped a stable 4.0.2.

File: src/System.IO.FileSystem/ref/4.0.2/project.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "dependencies": {
+    "System.Runtime": "4.0.0",
```
Comment (weshaggard) [inline src/System.IO.FileSystem/ref/4.0.2/project.json]: This should only have a reference to System.IO.FileSystem 4.0.1

File: src/System.IO.FileSystem/pkg/System.IO.FileSystem.pkgproj (hunk 1 above)
Comment (ericstj) [inline src/System.IO.FileSystem/pkg/System.IO.FileSystem.pkgproj]: Yes, and by convention just use 4.0 since we won't have more than one 4.0.x version.

Comment (JeremyKuhne) [inline src/System.IO.FileSystem/pkg/System.IO.FileSystem.pkgproj]: Ok, thanks.

File: src/System.IO.FileSystem/ref/4.0.2/project.json (hunk 2 above)
Comment (JeremyKuhne) [inline src/System.IO.FileSystem/ref/4.0.2/project.json]: Thanks, I'll fix it.

**Comment by stephentoub** [comment]
This is covered by #10022 and can be closed, right?

**Comment by JeremyKuhne** [comment]
Should be handled by https://github.com/dotnet/corefx/commit/62334804aa655a250838eefed68553f357ebec5f.

**Pull request closed without merging**